File: ntucoder/5533.cpp
#include <iostream>
#include <vector>
using namespace std;

#define ll long long

int main(){
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);
    int n, v;
    cin >>n>>v;

    ll temp;
    ll acc[n+1];
    acc[0] = 0;
    for ( int i =1 ; i <= n; ++i){
        cin >> temp;
        acc[i] = acc[i-1] + temp;
    }

    ll res = 0;
    int mark = 1;
    for (int i= 1; i <=n ; ++i){
        temp = acc[i-1] + v;
        while (mark <= n && acc[mark] < temp){
            mark++;
        }
        if (mark <= n){
            res += n + 1 - mark;
        } else{
            break;
        }
    }
    cout << res;
    return 0;
}

File: ntucoder/5533.py
n, v = list(map(int, input().split()))

acc = [0]
for i in range(n):
    acc.append(int(input()) + acc[-1])

res = 0
mark = 1
for i in range(1, len(acc)):
    while mark < len(acc) and acc[mark] < acc[i-1] + v:
        mark += 1
        
    if mark < len(acc):
        res += len(acc) - mark
    else:
        break

print(res)
